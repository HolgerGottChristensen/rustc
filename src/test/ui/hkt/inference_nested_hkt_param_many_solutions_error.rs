// fail because there are too many solutions for I and L
// check-fail
fn test<I<%J>, L<%K>>(input: I<L<u32>>) {
}

fn main() {
    test(Some(Some(5u32))); //~ ERROR too many possible types to infer HKT parameters [E10000]
}
