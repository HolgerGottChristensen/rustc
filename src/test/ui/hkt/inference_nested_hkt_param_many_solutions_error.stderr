error[E10000]: too many possible types to infer HKT parameters
  --> $DIR/inference_nested_hkt_param_many_solutions_error.rs:7:5
   |
LL |     test(Some(Some(5u32)));
   |     ^^^^
   |     |
   |     cannot infer for HKT param `"I"`
   |     help: try annotating the function call with type: `std::option::Option<std::option::Option<%K>>`

error[E10000]: too many possible types to infer HKT parameters
  --> $DIR/inference_nested_hkt_param_many_solutions_error.rs:7:5
   |
LL |     test(Some(Some(5u32)));
   |     ^^^^
   |     |
   |     cannot infer for HKT param `"L"`
   |     help: try annotating the function call with type: `std::option::Option<std::option::Option<%K>>`

error: aborting due to 2 previous errors

